### PR TITLE
Prepare for 7.3.0

### PR DESCRIPTION
**Pull request by bperseghetti**
# 🎈 Release

Preparation for 7.3.0 release.

## Checklist
- [x] Asked team if this is a good time for a release
- [x] There are no changes to be ported from the previous major version
- [ ] No PRs targeted at this major version are close to getting in
- [x] Bumped minor for new features, patch for bug fixes
- [x] Updated changelog
- [ ] Updated migration guide (as needed)
- [ ] Link to PR updating dependency versions in appropriate repository in [gazebo-release](https://github.com/gazebo-release) (as needed): <LINK>

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by azeey — APPROVED** [review]
Comparison to 7.2.0: https://github.com/gazebosim/gz-physics/compare/gz-physics7_7.2.0...gz-physics7

File: Changelog.md [hunk 1]
```diff
@@ -1,5 +1,52 @@
...
+    * [Pull request #606](https://github.com/gazebosim/gz-physics/pull/606)
+
+1. backport bullet-featherstone solver iters
+    * [Pull request #619](https://github.com/gazebosim/gz-physics/pull/619)
+
+1. bullet-featherstone: fix SetWorldPose with off-diagonal moment of inertia
+    * [Pull request #623](https://github.com/gazebosim/gz-physics/pull/623)
+
+1. Revert "Disable check in DetachableJointTest, CorrectAttachmentPoints for dartsim plugin on macOS (#613)"
+    * [Pull request #613)" (#615](https://github.com/gazebosim/gz-physics/pull/613)" (#615)
```
Comment (scpeters) [inline Changelog.md:48]:
```suggestion
1. Revert "Disable check in DetachableJointTest, CorrectAttachmentPoints for dartsim plugin on macOS (#613)"
    * [Pull request #615](https://github.com/gazebosim/gz-physics/pull/615)
```

**Comment by bperseghetti** [comment]
@azeey did the windows CI machine die?

**Comment by scpeters** [comment]
> @azeey did the windows CI machine die?

I just restarted the job. We've been having trouble with the windows machines disconnecting

**Review by scpeters — APPROVED** [review]

**Comment by azeey** [comment]
Windows is having a hard time today. It passed in https://build.osrfoundation.org/job/gz_physics-pr-win/276/ so I'll go ahead and merge this.

**Comment by peci1** [comment]
There's a possible "fixup" for this release (or rather a piggyback request) in https://github.com/gazebosim/gz-physics/pull/641#issuecomment-2190018675 .

**Pull request merged**